homeassistant/components/linknlink/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The linknlink integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .device import LinknLinkDevice
+from .heartbeat import LinknLinkHeartbeat
+
+
+@dataclass
+class linknlinkData:
+    """Class for sharing data within the linknlink integration."""
+
+    devices: dict[str, LinknLinkDevice] = field(default_factory=dict)
+    platforms: dict = field(default_factory=dict)
+    heartbeat: LinknLinkHeartbeat | None = None
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a linknlink device from a config entry."""
+    if not hass.data.get(DOMAIN):
+        hass.data[DOMAIN] = linknlinkData()
+    data: linknlinkData = hass.data[DOMAIN]
+
+    device = LinknLinkDevice(hass, entry)
+    if not await device.async_setup():
+        return False
```

Comment:
Should we log something? Does this mean the device is off forever? Why don't we raise ConfigEntryNotReady?